### DISCUSSION TITLE
How to Create a Dropdown in Statamic CP that Fetches Data from a Laravel Eloquent Model Using Runway

### DISCUSSION THREAD

**Discussion by ArlindKerqeli** [statamic-rad-pack/runway#475] [Help] [answered]
### Description

Hello,

I’m trying to create a dropdown select field within the Statamic Control Panel that dynamically populates its options from a Laravel Eloquent model. I've followed the Statamic documentation and understand that Runway is the recommended addon for integrating Eloquent models with Statamic.

Despite this, I'm facing difficulties in setting up the dropdown to correctly retrieve data from my Eloquent model. I've reached out to Statamic support and was pointed towards using Runway, but I haven’t had success in implementing it. The dropdown I’m aiming for should function similarly to the default 'Author' dropdown that Statamic provides.

Has anyone achieved this or could provide guidance on setting up a dropdown select field populated with Eloquent data using Runway?

What I've Tried:

1. Following the Runway documentation to set up a basic integration with my Eloquent model.
2. Attempting to add a select field type in the appropriate blueprint, with the expectation that it would automatically fetch data from my Eloquent model.

I'm a bit stuck and would really appreciate any advice or examples from others who have successfully implemented a similar feature.

Thank you for your time and assistance.

### Environment

$ php please support:details

Environment
Application Name: qSkills
Laravel Version: 9.52.16
PHP Version: 8.1.26
Composer Version: 2.6.6
Environment: local
Debug Mode: ENABLED
URL: localhost
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / daily
Mail: smtp
Queue: sync
Scout: meilisearch
Session: file

Statamic
Addons: 1
Antlers: regex
Sites: 3 (English, German, French)
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.55.0 PRO

Statamic Addons
statamic-rad-pack/runway: 6.6.2

**Comment by duncanmcclean** [comment]
Have you tried using Runway's [Belongs To](https://runway.duncanmcclean.com/fieldtypes#content-belongs-to) fieldtype?

**Reply by ArlindKerqeli** [reply]
I've tried it but with no success. I'm having an error, you can check the image uploaded below. 


<img width="1263" alt="Screenshot 2024-05-07 at 09 09 43" src="https://github.com/statamic-rad-pack/runway/assets/39010356/957a6286-3153-4e4f-a1e8-b1238148b2d0">



Also this is the code from the /config/runway.php:


` 
'resources' => [
         \App\Models\THotel::class => [
             'name' => 'Hotels',
             'route' => '/hotel/{{ id }}',
             'template' => 'front.hotel',
             'layout' => 'main',
         ],
    ],
    `

**Reply by duncanmcclean** [reply, marked as answer]
What does the resource's blueprint look like? Does it have any listable fields?

**Reply by ArlindKerqeli** [reply]
Thank you, I've fixed it now it works perfectly fine.
